src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1462,13 +1462,23 @@ private ConditionalOperation GenerateNullCheckForParameter(IParameterSymbol para
                 var nullableHasValue = ((IMethodSymbol)_compilation.CommonGetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue))?.Construct(parameter.Type);
                 if (nullableHasValue is null)
                 {
-                    conditionOp = new InvalidOperation(
-                        ImmutableArray.Create((IOperation)OperationCloner.CloneOperation(paramNameLiteral)),
+                    conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvalidOperation(
+                        ImmutableArray.Create((IOperation)OperationCloner.CloneOperation(paramReference)),
```

Comment:
Does the paramReference have to be cloned? It doesn't have to be in the else branch, so I assume it doesn't here.